src/librustdoc/doctree.rs

```diff
@@ -11,8 +11,10 @@ crate struct Item<'hir> {
     crate hir_item: &'hir hir::Item<'hir>,
     /// the explicit renamed name
     crate renamed_name: Option<Symbol>,
+    /// the [`Namespace`] this Item belongs to
+    crate namespace: Option<hir::def::Namespace>,
```

Comment:
Why store this on the item itself? Calculating it in `push_item` would use a lot less memory.